content/en/blog/_posts/2021-11-02-csi-migration-status.md

```diff
@@ -0,0 +1,130 @@
+
+`CSIMigration{provider}Complete` was introduced before as a supplementary feature gate once CSI migration is enabled on all of the nodes. This flag will de-register the in-tree storage plugin and directly translate the traffic to CSI driver without any checking if CSI migration is enabled on the node.
+
+While this feature gate is a great helper, we would also like a feature gate that can disable in-tree storage plugin without CSI migration enabled. For example, we would like to disable EBS storage plugin on a GKE cluster because it does not have ability to provision EBS volume. Thus, a new feature flag `InTreePlugin{vendor}Unregister` was introduced in kubernetes v1.21.
+
+`InTreePlugin{vendor}Unregister` is a standalone feature gate that can be enabled and disabled out of CSI Migration scope. The name speaks for itself, when enabled, the component will not register the specific in-tree storage plugin to the supported list. If the cluster operator only enables this flag, end users will get an error from PVC saying it cannot find the plugin when the plugin is used. The cluster operator may want to enable this regardless of CSI Migration if they do not want to support the legacy in-tree APIs and only support CSI moving forward.
+
+### Observability
+
+Kubernetes v1.21 introduced [metrics](https://github.com/kubernetes/kubernetes/issues/98279) for tracking CSI migration.
+You can use these metrics to observe how your cluster is using storage services and whether access to that storage is using the legacy in-tree driver or its CSI-based replacement.
+
+| Components                                   | Metrics                            | Notes                                                                                                                                                                                  |
+| -------------------------------------------- | ---------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Kube-Controller-Manager                      | storage_operation_duration_seconds | A new label `migrated` is added to the metric to indicate whether this storage operation is a CSI migration operation(`True` for enabled and `False` for not enabled).                 |
+| Kubelet                                      | csi_operations_seconds             | The new metric exposes labels including `driver_name`, `method_name`, `grpc_status_code` and `migrated`. The meaning of these labels is identical to `csi_sidecar_operations_seconds`. |
+| CSI Sidecars(provisioner, attacher, resizer) | csi_sidecar_operations_seconds     | A new label `migrated` is added to the metric to indicate whether this storage operation is a CSI migration operation(`True` for enabled and `False` for not enabled).                 |
```

Comment:
OpenMetrics [doesn't have](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#data-model) native Boolean values. Let's clarify that the values are `0` and `1` (or `0.0` and `1.1`) at the metrics endpoint) - assuming I'm right here.

Reply:
I took another look and it seems we were using string value true and false. Updated the doc